src/core/border_router/routing_manager.cpp

```diff
@@ -449,18 +449,29 @@ void RoutingManager::EvaluateOmrPrefix(OmrPrefixArray &aNewOmrPrefixes)
         // The `aNewOmrPrefixes` remains empty if we fail to publish
         // the local OMR prefix.
     }
-    else if (publishedLocalOmrPrefix != nullptr && smallestOmrPrefix != publishedLocalOmrPrefix)
+    else
     {
-        otLogInfoBr("EvaluateOmrPrefix: There is already a smaller OMR prefix %s in the Thread network",
-                    smallestOmrPrefix->ToString().AsCString());
+        OT_ASSERT(smallestOmrPrefix != nullptr);
 
-        UnpublishLocalOmrPrefix();
+        if (*smallestOmrPrefix == mLocalOmrPrefix)
+        {
+            IgnoreError(PublishLocalOmrPrefix());
```

Comment:
Better logging the error but not ignoring it.

Reply:
We are already logging the error in PublishLocalOmrPrefix
.